consensus/view_change.go

```diff
@@ -624,14 +633,37 @@ func (consensus *Consensus) onNewView(msg *msg_pb.Message) {
 			preparedBlock.Epoch(), preparedBlock.Hash(), preparedBlock.NumberU64(), preparedBlock.Header().ViewID().Uint64())
 		groupID := []nodeconfig.GroupID{
 			nodeconfig.NewGroupIDByShardID(nodeconfig.ShardID(consensus.ShardID))}
-		for _, key := range consensus.priKey {
+
+		priKeys := []*bls.PrivateKeyWrapper{}
+		for i, key := range consensus.priKey {
 			if !consensus.IsValidatorInCommittee(key.Pub.Bytes) {
 				continue
 			}
+			priKeys = append(priKeys, &consensus.priKey[i])
+			if !consensus.MultiSig {
+				network, err := consensus.construct(
```

Comment:
same thing as validator.go, if we can refactor the code here. if it is too complicated, then leave it.